liborbum/src/Controller/Ee/Vpu/Vu/Interpreter/CVuInterpreter_FLOAT.cpp

```diff
@@ -779,50 +1517,106 @@ void CVuInterpreter::MINIbc_3(VuUnit_Base* unit, const VuInstruction inst)
+
+    FpuFlags flags;
+
+    acc.write_float(VuVectorField::X, to_ps2_float(
+                                          fs.read_float(VuVectorField::Y) * ft.read_float(VuVectorField::Z), flags));
+    unit->mac.update_vector_field(VuVectorField::X, flags);
+
+    acc.write_float(VuVectorField::Y, to_ps2_float(
+                                          fs.read_float(VuVectorField::Z) * ft.read_float(VuVectorField::X), flags));
+    unit->mac.update_vector_field(VuVectorField::Y, flags);
+
+    acc.write_float(VuVectorField::Z, to_ps2_float(
+                                          fs.read_float(VuVectorField::X) * ft.read_float(VuVectorField::Y), flags));
+    unit->mac.update_vector_field(VuVectorField::Z, flags);
 }
 
 void CVuInterpreter::OPMSUB(VuUnit_Base* unit, const VuInstruction inst)
 {
-    // TODO: Implement.
-#if defined(BUILD_DEBUG)
-    BOOST_LOG(Core::get_logger()) << boost::format("(%s, %d) OPMSUB: Not implemented.") % __FILENAME__ % __LINE__;
-#else
-    throw std::runtime_error("OPMSUB: Not implemented.");
-#endif
+    SizedQwordRegister& fd = unit->vf[inst.fd()];
+    SizedQwordRegister& fs = unit->vf[inst.fs()];
+    SizedQwordRegister& ft = unit->vf[inst.ft()];
+    SizedQwordRegister& acc = unit->acc;
+
+    FpuFlags flags;
+    fd.write_float(VuVectorField::X, to_ps2_float(
+                                         acc.read_float(VuVectorField::X) - fs.read_float(VuVectorField::Y) * ft.read_float(VuVectorField::Z), flags));
+    unit->mac.update_vector_field(VuVectorField::X, flags);
+
+    fd.write_float(VuVectorField::Y, to_ps2_float(
+                                         acc.read_float(VuVectorField::Y) - fs.read_float(VuVectorField::Z) * ft.read_float(VuVectorField::X), flags));
+    unit->mac.update_vector_field(VuVectorField::Y, flags);
+
+    fd.write_float(VuVectorField::Z, to_ps2_float(
+                                         acc.read_float(VuVectorField::Z) - fs.read_float(VuVectorField::X) * ft.read_float(VuVectorField::Y), flags));
+    unit->mac.update_vector_field(VuVectorField::Z, flags);
 }
 
 void CVuInterpreter::DIV(VuUnit_Base* unit, const VuInstruction inst)
 {
-    // TODO: Implement.
-#if defined(BUILD_DEBUG)
-    BOOST_LOG(Core::get_logger()) << boost::format("(%s, %d) DIV: Not implemented.") % __FILENAME__ % __LINE__;
-#else
-    throw std::runtime_error("DIV: Not implemented.");
-#endif
+    SizedWordRegister& q = unit->q;
+    SizedQwordRegister& fs = unit->vf[inst.fs()];
+    SizedQwordRegister& ft = unit->vf[inst.ft()];
+
+    FpuFlags flags;
+    const float result = to_ps2_float(fs.read_float(inst.fsf()) / ft.read_float(inst.ftf()), flags);
+
+    unit->status.clear_flags();
+    
+    if (ft.read_float(inst.ftf()) == 0.0f) 
+    {
+        if (fs.read_float(inst.ftf()) == 0.0f) {
+            unit->status.insert_field(VuUnitRegister_Status::I, 1);
+        } 
+        else 
+        {
+            unit->status.insert_field(VuUnitRegister_Status::D, 1);
```

Comment:
the sticky flag is not set by doing this - use `status.set_d_flag_sticky()`

Reply:
Yeah I just noticed that while updating the code, will fix it promptly.